src/mast/mod.rs

```diff
@@ -496,7 +496,7 @@ fn monomorphize_expr<B: Backend>(
             let lhs_typ = lhs_mono.typ.unwrap();
             let rhs_typ = rhs_mono.typ.unwrap();
 
-            if !lhs_typ.same_as(&rhs_typ) {
+            if !lhs_typ.match_expected(&rhs_typ, true) {
```

Comment:
Same as the other comment, This may require another flag to ensure the types are resolved types and not generic involved.